Core/include/Acts/Utilities/Result.hpp

```diff
@@ -71,40 +96,29 @@ class Result {
    * an error, and a value will be a value.
    * @note If T and E are ambigious, use the `success` and `failure` static
    * factory methods.
-   * @tparam T2 Type of the potential assignment
+   * @tparam U Type of the potential assignment
    * @param value The potential value, could be an actual valid value or an
    * error.
    */
-  template <
-      typename T2, typename _E = E, typename _T = T,
-      typename = std::enable_if_t<
-          (!std::is_same_v<_T, _E> && !std::is_constructible_v<_T, _E> &&
-           !std::is_convertible_v<_T, _E> && !std::is_constructible_v<_E, _T> &&
-           !std::is_convertible_v<_E, _T> &&
-           !(std::is_convertible_v<T2, _T> && std::is_convertible_v<T2, _E>))>>
-  Result(T2 value) noexcept
-      : m_var(std::conditional_t<std::is_convertible_v<T2, _T>, T, E>{
-            std::move(value)}) {}
+  template <typename U, typename = EnableIfConstructibleFrom<U>>
```

Comment:
I'm not sure this SFINAE's correctly if you don't pull in the class template parameter `T` EXPLICITLY into the method template.

Reply:
Good question. This seems to work as intended for the unit tests we have. Do you think you can come up with an example where it does not? Since `EnableIfConstructibleFrom` is defined as dependent on `T`, I was assuming that it would.

Reply:
I think you'll only see the problem if this does NOT match. SFINAE means it should be removed from overload resolution, and you'll get a compile error that says something like "no overload matched". With this, it will give you a different error, which I'd guess is a lot more obtuse. In the end it doesn't matter all that much, since it won't compile either way, but would argue 'proper' SFINAE is still desireable.

Reply:
> With this, it will give you a different error, which I'd guess is a lot more obtuse.

That is what I would like to have an example for. Just to clarify, this change was done to fix a compiler error with the previous implementation. The compiler complained that using it with some types was resulting in an implicit narrowing conversion.